crates/macro-support/src/lib.rs

```diff
@@ -104,9 +104,70 @@ pub fn expand_class_marker(
     Ok(tokens)
 }
 
+/// Describes how to rename the fields, methods, and classes in the generated
+/// bindings.
+///
+/// This design heavily mimics the `#[serde(rename_all = "...")]` attribute.
+#[derive(Debug, Clone, Copy)]
+enum RenameRule {
+    None,
+    CamelCase,
+}
+impl RenameRule {
+    fn snake_case_to_camel_case(name: String) -> String {
+        let mut camel = String::new();
+        let mut capitalize = false;
+        for c in name.chars() {
+            if c == '_' {
+                capitalize = true;
+            } else if capitalize {
+                camel.push(c.to_ascii_uppercase());
+                capitalize = false;
+            } else {
+                camel.push(c);
+            }
+        }
+        camel
+    }
+
+    fn rename_snake_case(self, name: String) -> String {
+        match self {
+            RenameRule::None => name.to_string(),
+            RenameRule::CamelCase => Self::snake_case_to_camel_case(name),
+        }
+    }
+
+    /// Applies the rule to the field of a struct or enum variant data.
+    fn apply_to_field(self, name: String) -> String {
+        self.rename_snake_case(name)
+    }
+    /// Applies the rule to the method of a struct or enum.
+    fn apply_to_method(self, name: String) -> String {
+        self.rename_snake_case(name)
+    }
```

Comment:
Why are we separating those?

Reply:
Lets not overextend, this can always be expanded when we actually get there.

Reply:
I replaced with all with a single `snake_case_to_camel_case` function.